elements/ethernet/etherpausesource.cc

```diff
@@ -97,12 +104,18 @@ EtherPauseSource::cleanup(CleanupStage)
 void
 EtherPauseSource::run_timer(Timer *)
 {
-    if (Packet *p = _packet->clone()) {
-	++_count;
-	output(0).push(p);
+    Packet *p = _packet->clone();
+    if (p) {
+        ++_count;
+    #if HAVE_BATCH
+        output_push_batch(0, PacketBatch::make_from_packet(p));
```

Comment:
Same here

Reply:
Sure I can replace:
#if HAVE_BATCH
with
if (in_batch_mode == BATCH_MODE_YES) {

but I believe it is not necessary in this case.
Both elements are source elements (i.e., they have on input ports) and both of them have:
#if HAVE_BATCH
    in_batch_mode = BATCH_MODE_YES;
#endif

in their constructor, so the path begins from these elements and in_batch_mode is always BATCH_MODE_YES.

If you still believe I am wrong, I can push the changes :)

Reply:
You are correct. I forgot they were source and therefore always in batch mode.